apollo-router/src/spec/query/change.rs

```diff
@@ -0,0 +1,611 @@
+                        self.hash_directive(directive);
+                    }
+                }
+            }
+            ExtendedType::InputObject(o) => {
+                for directive in &o.directives {
+                    self.hash_directive(&directive.node);
+                }
+
+                for (name, ty) in &o.fields {
+                    if ty.default_value.is_some() {
+                        name.hash(self);
+                        self.hash_input_value_definition(&ty.node);
+                    }
+                }
+            }
+        }
+    }
+
+    fn hash_type(&mut self, t: &ast::Type) {
+        match t {
+            schema::Type::Named(name) => self.hash_type_by_name(name.as_str()),
+            schema::Type::NonNullNamed(name) => {
+                "!".hash(self);
+                self.hash_type_by_name(name.as_str())
+            }
+            schema::Type::List(t) => {
+                "[]".hash(self);
+                self.hash_type(t);
+            }
+            schema::Type::NonNullList(t) => {
+                "[]!".hash(self);
+                self.hash_type(t);
+            }
+        }
+    }
+
+    fn hash_input_value_definition(&mut self, t: &Node<ast::InputValueDefinition>) {
+        self.hash_type(&t.ty);
+        for directive in &t.directives {
+            self.hash_directive(directive);
+        }
+        if let Some(value) = t.default_value.as_ref() {
+            self.hash_value(value);
+        }
+    }
+
+    fn hash_entities_operation(&mut self, node: &ast::OperationDefinition) -> Result<(), BoxError> {
+        use crate::spec::query::traverse::Visitor;
+
+        if node.selection_set.len() != 1 {
+            return Err("invalid number of selections for _entities query".into());
+        }
+
+        match node.selection_set.first() {
+            Some(Selection::Field(field)) => {
+                if field.name.as_str() != "_entities" {
+                    return Err("expected _entities field".into());
+                }
+
+                "_entities".hash(self);
+
+                for selection in &field.selection_set {
+                    match selection {
+                        Selection::InlineFragment(f) => {
+                            match f.type_condition.as_ref() {
+                                None => {
+                                    return Err("expected type condition".into());
+                                }
+                                Some(condition) => self.inline_fragment(condition.as_str(), f)?,
+                            };
+                        }
+                        _ => return Err("expected inline fragment".into()),
+                    }
+                }
+                Ok(())
+            }
+            _ => Err("expected _entities field".into()),
+        }
+    }
+}
+
+impl<'a> Hasher for QueryHashVisitor<'a> {
+    fn finish(&self) -> u64 {
+        //self.hasher.finalize()
+        todo!()
```

Comment:
Is it a mistake ?

Reply:
it is not used, the real hash is generated elsewhere